### DISCUSSION TITLE
Pipeline does not behave as I'd expect 🤔

### DISCUSSION THREAD

**Discussion by testerez** [vercel/turborepo#902] [Help] [answered]
Can someone please explain to me why `b:test` is executed when I run:
```
> yarn turbo run test --scope=a --include-dependencies --force

b:build: cache bypass, force executing ec273d1a50e8afb3
b:test: cache bypass, force executing 5813cbc82183a8c5
b:build: $ echo building b
b:test: $ echo testing b
b:build: building b
b:test: testing b
a:test: cache bypass, force executing ea636699505bd1da
a:test: $ echo testing a
a:test: testing a

 Tasks:    3 successful, 3 total
Cached:    0 cached, 3 total
  Time:    527ms 
```

My pipeline is: 
```
  "pipeline": {
    "build": {
      "dependsOn": ["^build"]
    },
    "test": {
      "dependsOn": ["^build"]
    }
  }
```

Here I would expect only `a:test` to be selected along with `b:build` since it's a dependancy. But I don't understand why `b:test` is also executed.

The full test project can be found here: https://github.com/testerez/test-turbo

**Comment by anthonyshew** [comment, marked as answer]
The `^` character asks your pipeline to run the task in the topological graph of your repository. `^topo` means "Run the tests scripts of dependents before dependencies."
